packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -651,8 +651,12 @@ mixin LocalHistoryRoute<T> on Route<T> {
 
 class _DismissModalAction extends DismissAction {
   @override
-  Object invoke(DismissIntent intent) {
-    return Navigator.of(primaryFocus!.context!)!.maybePop();
+  Object? invoke(DismissIntent intent) {
```

Comment:
I _think_ we can get away without calling `notifyActionListeners` since the `Actions` instance that holds the `DismissModalAction` is inserted into the hierarchy for a specific modal route, and that route's `isBarrierDismissible` property is immutable.  Therefore, the enabled state of the action is effectively immutable as well.  wdyt?

Reply:
Hmm, scratch that - we reuse a static action, and it uses the context of the primary focus.  We could change that to instantiate a unique `DismissModalAction` for every modal route -- and make each instantiation use the context in which it was created rather than the context of the focus?

Reply:
Alternatively, the most "correct" thing to do would be to re-check if the modal barrier is dismissible every time the primary focus changes, but I worry that that's overkill (and not sure if it's possible to be notified when the primary focus changes).

Reply:
Ok, I updated it to create a new actions map (with a new action instance) for every modal barrier, thus allowing us to never notify.  PTAL.

Reply:
It is possible to listen to focus changes: the `FocusManager` is a `ChangeNotifier` for that reason.